ledger/participant-integration-api/src/main/scala/platform/store/dao/events/EventsTable.scala

```diff
@@ -99,14 +99,21 @@ object EventsTable {
       // The identifiers of all visible events in this transactions, preserving
       // the order in which they are retrieved from the index
       val visible = events.map(_.event.eventId)
-      val visibleSet = visible.toSet
+      val visibleOrder = visible.view.zipWithIndex.toMap
```

Comment:
establishing order from the order of the events: that is preserved correctly as indexing (as opposed to extracting order from the eventId string, to prevent formation of tight coupling to EventId string rendering logic)